firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorMiddleware.swift

```diff
@@ -0,0 +1,31 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Redux
+import Shared
+import Common
+
+final class PasswordGeneratorMiddleware: FeatureFlaggable {
+    lazy var passwordGeneratorProvider: Middleware<AppState> = { state, action in
+        let windowUUID = action.windowUUID
+        guard let currentTab = (action as? PasswordGeneratorAction)?.currentTab else { return }
+        switch action.actionType {
+        case PasswordGeneratorActionType.showPasswordGenerator:
+                self.showPasswordGenerator(tab: currentTab, windowUUID: windowUUID)
+        default:
+            break
+        }
+    }
+
+    private func showPasswordGenerator(tab: Tab, windowUUID: WindowUUID) {
+        // TODO - Business Logic to be added (tab is a necessary part of future business logic)
```

Comment:
Can you add a ticket number to this TODO so it can be tracked later if the comment is never removed.